src/OutputWriters/jld2_output_writer.jl

```diff
@@ -190,7 +190,7 @@ function JLD2OutputWriter(model, outputs; filename, schedule,
 
     initialize_jld2_file!(filepath, init, jld2_kw, including, outputs, model)
     
-    mask_value = mask_immersed is Number ? eltype(model.grid)(mask_immersed) : mask_immersed
+    mask_value = mask_immersed isa Number ? eltype(model.grid)(mask_immersed) : mask_immersed
```

Comment:
I don't think this matters. It would be converted inside `mask_immersed!`.

Also I think it's better to keep the same name as the kwarg because its simpler and easier to understand. Then there's a property `output_writer.mask_immersed` which should be the input value.